### DISCUSSION TITLE
Unable to Load Dynamic Stylesheets in Next.js App Router

### DISCUSSION THREAD

**Discussion by towhidstack** [vercel/next.js#59573] [App Router]
I am currently working on the development of a Headless CMS for WordPress using Next.js. In Next.js Version 12, I successfully implemented the loading of dynamic stylesheets using the Head component, as demonstrated in the following code snippet:

```jsx
const elementorDynamicCssLinks = elementorPostCssLinkData(pageId)

<Head>
  {data &&
    settings?.config?.hasElementorSupport &&
    elementorDynamicCssLinks.length &&
    elementorDynamicCssLinks.map((link: any) => (
      <link
        key={link?.id}
        rel='stylesheet'
        id={link?.id}
        href={link?.link}
        media='all'
      />
    ))}
</Head>
```

This approach allowed me to efficiently load dynamic stylesheets based on the page's content. However, upon migrating to Next.js Version 14, I am unable to find a suitable method to achieve the same dynamic stylesheet loading functionality Inside App Router. Despite exploring the documentation and community resources, I have not come across a clear solution for this specific use case in the latest version.

I would greatly appreciate any guidance, examples, or documentation that can assist me in adapting this dynamic stylesheet loading mechanism to Next.js Version 14. If there are new APIs or changes in the framework that affect this functionality, I believe clarifications or adjustments would benefit the community using Next.js for WordPress Headless CMS projects.

Thank you for your support in addressing this challenge.

**Comment by rnnyrk** [comment]
Think this discussion is related: https://github.com/vercel/next.js/discussions/49744
